### DISCUSSION TITLE
How to view the contents of rtccalls.db and rtccontacts.db files?

### DISCUSSION THREAD

**Discussion by NCHU611A** [mehrvarz/webcall#34] [Q&A] [answered]
Hi,
I want to see what information is recorded in **rtccalls.db**, **rtccontacts.db**, **rtchashedpw.db**, **rtcnotif.db**, **rtcsig.db**.

Is there a way for me to see the contents of these files?

If so, what software should I use to open them?

Because I can't see the content inside by double-clicking them directly.

Thank you.

**Comment by nscomputing** [comment, marked as answer]
These files are BoltDB files ( https://dbdb.io/db/boltdb ). 

To view contents of these files you can use  this command line  browser for BoltDB Files: https://github.com/br0xen/boltbrowser

**Reply by nscomputing** [reply]
The current Bolt DB project url: https://github.com/etcd-io/bbolt

**Comment by mehrvarz** [comment]
rtccalls.db contains two buckets (tables): dbWaitingCaller, dbMissedCalls
rtcsig.db contains three buckets: dbRegisteredIDs, dbBlockedIDs and dbUserBucket
rtccontacts.db contains one bucket: dbContactsBucket
rtchashedpw.db contains one bucket: dbHashedPwBucket
rtcnotif.db contains one bucket: dbSentNotifTweets
(rtcnotif.db / dbSentNotifTweets are not being fed any data at this time.)

You can find these relations also at the top of main.go (right under imports).

No tools exist, currently, to read/view these db-files. While such tools could be created, I am operating the server without any such tools. And I am not missing anything.

**Reply by NCHU611A** [reply]
Got it.
Thank you !

**Comment by nscomputing** [comment]
With Boltbrowser you can browse, viev , edit, create, delete pair/bucket. 

Boltbrowser help screen

![boltbrowser-Screenshot_20230109_172154](https://user-images.githubusercontent.com/32056985/211251781-01af7da6-70ac-4a67-bbd1-1c340a24030a.png)

View/edit key/value pair in rtchashedpw.db


![rtchashedpw-db-view-edit-value-Screenshot_20230109_171749](https://user-images.githubusercontent.com/32056985/211252222-abca1437-7c43-4cb9-b609-817b266c392a.png)

**Reply by NCHU611A** [reply]
Hi, I execute "`go get github.com/br0xen/boltbrowser`" directly on the terminal and the result shows this message.

> go: go.mod file not found in current directory or any parent directory.
	'go get' is no longer supported outside a module.
	To build and install a command, use 'go install' with a version,
	like 'go install example.com/cmd@latest'
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.

Can you teach me how to install it ?

My Golang version : `go version go1.19.4 linux/amd64`

My computer operating system : Ubuntu 20.04 LTS

Thank you !!

**Comment by nscomputing** [comment]
Please try to execute "go install github.com/br0xen/boltbrowser@latest" . 
This worked in my environment: Debian 11, go1.19.3 linux/amd64.

If this doesn't work in your environment, you can download pre-built binary for Linux 64-bit: https://git.bullercodeworks.com/attachments/29367198-79f9-4fb3-9a66-f71a0e605006

**Reply by NCHU611A** [reply]
After I executed the "`go install github.com/br0xen/boltbrowser@latest`" command on the terminal, it did not report any errors.
Then I create a new folder and put the .db file in it.
Then open the terminal in this folder path.
Then I execute the "`boltbrowser myfile.db`" command but the terminal reports "`boltbrowser: command not found`"

**Reply by nscomputing** [reply]
It is important that you use full paths of both, the executable (in this case boltbrowser) and the database file (in this case myfile.db).

Example:

1. If executable boltbrowser is located in folder /home/your_username/go/bin 
2. And myfile.db is located in folder /home/your_username/dbfiles/myfile.db

Please try to execute these commands:

1. cd dbfiles
2. ~/go/bin/boltbrowser myfile.db

**Reply by NCHU611A** [reply]
I executed it successfully !
I can see the contents of "`myfile.db`" now.

Thank you!!

I have a less important problem now.

What command should be executed to make the terminal appear like the image below?
![211251781-01af7da6-70ac-4a67-bbd1-1c340a24030a](https://user-images.githubusercontent.com/114225240/211757081-8ec6c449-ca77-486f-a3aa-544137a89d6d.png)

**Reply by nscomputing** [reply]
Just press "?" to get this screen. Or "q" to quit program.

**Reply by NCHU611A** [reply]
I execute this command "`~/go/bin/boltbrowser ~/dbfiles/rtccalls.db`" on the terminal and press "**?**".
The terminal returned the screen of the following picture.

![1673528402219](https://user-images.githubusercontent.com/114225240/212073010-75eb8bdf-7767-4654-b497-74ad08b3fa58.jpg)

The picture below does not appear.

![211251781-01af7da6-70ac-4a67-bbd1-1c340a24030a](https://user-images.githubusercontent.com/114225240/212075539-d6e4763f-f0f1-4105-b2b8-71e2db90a60b.png)

Thank you!!

**Reply by nscomputing** [reply]
It works for me. 
Are you sure that you have typed "?" ?
Maybe you should press "shift - ? " on your keyboard to get "?" .

**Reply by NCHU611A** [reply]
OK! I found it.
Thank You !
